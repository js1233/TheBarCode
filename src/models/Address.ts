import DateTime from "./DateTime";

export interface Address {
  id: number;
  user_id: number;
  title: string;
  address: string;
  city: string;
  latitude: number;
  longitude: number;
  optional_note: string;
  created_at: DateTime;
  updated_at: DateTime;
  deleted_at: DateTime;
  post_code: string;
}
