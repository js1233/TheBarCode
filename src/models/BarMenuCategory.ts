import DateTime from "./DateTime";

export interface BarMenuCategory {
  id: number;
  name: string;
  image: string;
  type: string;
  created_at: DateTime;
  updated_at: DateTime;
  deleted_at: null;
}
