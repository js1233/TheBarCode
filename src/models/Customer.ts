type Customer = {
  id: number;
  full_name?: string;
  contact_number?: null;
};

export default Customer;
