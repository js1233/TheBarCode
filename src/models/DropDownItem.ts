export type DropDownItem = {
  value?: string;
  text?: string;
};
