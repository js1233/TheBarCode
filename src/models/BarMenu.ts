import { AppLog } from "utils/Util";
import { ModifierDetails } from "./api_responses/ModifierDetailsResponseModel";
import { BarMenuCategory } from "./BarMenuCategory";
import DateTime from "./DateTime";
import EFunnelType from "./enums/EFunnelType";
import EPosType from "./enums/EPosType";
import EProductGroupType from "./enums/EProductGroupType";
import ESupportedOrderType from "./enums/ESupportedOrderType";
import { Modifier, modifierTotalPrice } from "./Modifier";
import { Venue } from "./Venue";

export interface BarMenu {
  id: number;
  establishment_id: number;
  category_id: number;
  name: string;
  image?: string;
  unit: null;
  price: number;
  description: null | string;
  status: number;
  created_at: DateTime;
  updated_at: DateTime;
  deleted_at: null;
  segment_id: number;
  sequence: number;
  menu_type: EPosType;
  plu: null;
  takeaway_delivery: boolean;
  ref_id: number;
  is_push: number;
  dine_in_collection: boolean;
  related_menus_ids: number[] | null;
  group_type: EProductGroupType;
  sub_menu_ids: null | string;
  start_date: Date | null;
  end_date: Date | null;
  start_time: null | string;
  end_time: null | string;
  combo_discount: number | null;
  details: string;
  is_allergen: boolean;
  allergen_description?: string;
  allergen_icons?: string[];
  have_modifiers: boolean;
  category: BarMenuCategory;
  establishment: Venue;
  image_name: null | string;
  related_menus?: BarMenu[];
  quantity: number;
  total: number;
  cart_item_id?: number | null;
  comment: null;
  sub_menus?: BarMenu[];
  modifiers?: Modifier[];
  key?: () => string;
  exclusive_offer_id: number;
  deal_price?: number;
  bundle_offer_type?: EFunnelType;
  bundle_discount?: number;
  free_funnel_id?: number;
  currency_symbol?: string;
  modifiers_acc_quan?: [[Modifier]] | [[]];
  basic_bogo?: boolean;
  items?: [BarMenu]; //value come in case of bundle
  funnel_details?:
    | [
        {
          id: string;
          menus: {
            id: string;
            quantity: number;
            modifier_details?: [ModifierDetails];
          };
        }
      ]
    | [];
  modifier_details?: Modifier[][]; //comes in case of bundle bogo in order summary
  offer_type?: string;
}

export const barMenuTotalPrice = (menu: BarMenu): number => {
  return barMenuUnitPrice(menu) * menu.quantity;
};

export const calculateBundlePrice = (menu: BarMenu): number => {
  let priceContainer: number[] = [];
  menu.sub_menus?.map((subMenu) => {
    let total: number = 0;
    total = subMenu.price;
    subMenu.modifiers?.map((modifier) => {
      total += modifier.price * modifier.quantity;
    });
    priceContainer.push(total);
  });
  let largest =
    priceContainer.length > 0 ? Math.max.apply(0, priceContainer) : 0;
  return largest;
};

export const barMenuUnitPrice = (menu: BarMenu): number => {
  // menu.total is used in case of bundle
  let total = isBundleBogo(menu) ? menu.total ?? 0.0 : menu.price;
  menu.modifiers?.forEach((_item) => {
    total += modifierTotalPrice(_item);
  });

  menu.sub_menus?.forEach((item) => {
    total += barMenuUnitPrice(item);
  });

  if (isBundleBogo(menu)) {
    total = total / menu.quantity;
  }
  return total;
};

export const isMenuOffersMultiCategories = (menu: BarMenu) => {
  return (
    menu.menu_type === EPosType.BARCODE &&
    menu.dine_in_collection &&
    menu.takeaway_delivery
  );
};

export const supportedOrderTypes = (menu: BarMenu) => {
  if (menu.dine_in_collection && menu.takeaway_delivery) {
    return ESupportedOrderType.ALL;
  } else if (menu.dine_in_collection) {
    return ESupportedOrderType.DINE_IN_COLLECTION;
  } else if (menu.takeaway_delivery) {
    return ESupportedOrderType.TAKEAWAY_DELIVERY;
  } else {
    return ESupportedOrderType.ALL;
  }
};

export const getAlergensIcon = (item: string) => {
  switch (item) {
    case "milk":
      return require("assets/images/milk.png");
    case "brazil-nut":
      return require("assets/images/brazil-nut.png");
    case "celery":
      return require("assets/images/cellery.png");
    case "crustacens":
      return require("assets/images/crustacea.png");
    case "eggs":
      return require("assets/images/egg.png");
    case "fish":
      return require("assets/images/fish.png");
    case "lupin":
      return require("assets/images/lupine.png");
    case "mustard":
      return require("assets/images/mustard.png");
    case "peanut":
      return require("assets/images/peanut.png");
    case "sesame":
      return require("assets/images/sesame-seed.png");
    case "soya":
      return require("assets/images/soya.png");
    case "sulphurdioxide":
      return require("assets/images/sulphurdioxide.png");
    case "molluscs":
      return require("assets/images/molluscs.png");
    case "wheat":
      return require("assets/images/wheat.png");
    default:
      AppLog.log(() => "Image doesn't exist " + item);
      break;
  }
};

export const isBundleBogo = (menu: BarMenu) =>
  menu.group_type === EProductGroupType.BUNDLE ||
  menu.group_type === EProductGroupType.BOGO;

export const isBasicBogo = (menu: BarMenu) =>
  menu?.group_type === EProductGroupType.BOGO && menu.basic_bogo;
