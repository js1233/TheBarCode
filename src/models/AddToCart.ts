import { BarMenu } from "./BarMenu";
import DateTime from "./DateTime";
import { Venue } from "./Venue";

export type AddToCart = {
  id: number;
  user_id: number;
  establishment_id: number;
  order_id: null;
  created_at: DateTime;
  updated_at: DateTime;
  epos_type: string;
  cart_type: string;
  offer_type: null;
  redeem_type: null;
  establishment: Venue;
  menuItems: BarMenu[];
  total: number;
};
