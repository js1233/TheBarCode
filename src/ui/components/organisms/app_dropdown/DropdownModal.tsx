import {
  ImageStyle,
  Modal,
  Pressable,
  StyleProp,
  StyleSheet,
  View
} from "react-native";
import React from "react";
import { AppLabel } from "ui/components/atoms/app_label/AppLabel";
import { FlatListWithPb } from "ui/components/organisms/flat_list/FlatListWithPb";
import { usePreferredTheme } from "hooks";
import { DropDownItem } from "models/DropDownItem";
import EIntBoolean from "models/enums/EIntBoolean";
import { FONT_SIZE, SPACE } from "config/Dimens";
import Cross from "assets/images/ic_cross.svg";

export interface DropDownModalProps {
  isVisible: boolean;
  closeModal: () => void;
  items: DropDownItem[] | undefined;
  selectedItemCallback: (item: DropDownItem) => void;
  dropDownBgColor?: string;
  dialogCloseIconStyle?: StyleProp<ImageStyle>;
  selectedItemPosition: number;
  isLocked?: EIntBoolean;
}

export const DropdownModal = React.memo<DropDownModalProps>(
  ({
    isVisible,
    closeModal,
    items,
    selectedItemCallback,
    dropDownBgColor = null,
    selectedItemPosition
  }) => {
    const { themedColors } = usePreferredTheme();

    const getItemColor = (position: number): string => {
      if (selectedItemPosition === position) {
        return themedColors.interface["900"];
      } else {
        return themedColors.interface["700"];
      }
    };

    const renderItem = ({
      item,
      index
    }: {
      item: DropDownItem;
      index: number;
    }) => {
      return (
        <Pressable
          testID="dropdown-item-click"
          style={({ pressed }) => [{ opacity: pressed ? 0.5 : 1.0 }]}
          onPress={() => selectedItemCallback(item)}>
          <AppLabel
            text={item.value!}
            style={[styles.flatListItem, { color: getItemColor(index) }]}
          />
        </Pressable>
      );
    };

    return (
      <Modal
        testID="dropdown-modal"
        visible={isVisible}
        animationType="none"
        transparent={true}>
        <View style={styles.root}>
          <View style={styles.contentWrapper}>
            <View
              style={[
                styles.content,
                {
                  backgroundColor: dropDownBgColor
                    ? dropDownBgColor
                    : themedColors.interface["100"]
                }
              ]}>
              <FlatListWithPb
                shouldShowProgressBar={false}
                data={items}
                style={styles.flatList}
                renderItem={renderItem}
                showsVerticalScrollIndicator={false}
                initialNumToRender={items?.length}
                removeClippedSubviews={true}
                keyExtractor={(item, index) => index.toString()}
              />
            </View>
            <Pressable
              style={[
                styles.circleWrapper,
                { backgroundColor: themedColors.interface["100"] }
              ]}
              testID="close-modal"
              onPress={closeModal}>
              <View
                style={[
                  styles.circle,
                  {
                    borderColor: themedColors.interface["300"],
                    backgroundColor: themedColors.interface["200"]
                  }
                ]}>
                <Cross
                  width={15}
                  height={15}
                  fill={themedColors.interface["700"]}
                />
              </View>
            </Pressable>
          </View>
        </View>
      </Modal>
    );
  }
);

const styles = StyleSheet.create({
  root: {
    flex: 1,
    justifyContent: "center",
    backgroundColor: "rgba(90,94,94,0.7)"
  },
  //to handle auto height according to list
  contentWrapper: {
    flex: 0,
    height: "auto"
  },
  content: {
    borderRadius: 10,
    justifyContent: "center",
    marginHorizontal: 25,
    marginVertical: 50
  },
  circleWrapper: {
    top: 50,
    left: 25,
    padding: 10,
    justifyContent: "center",
    alignItems: "center",
    position: "absolute",
    borderRadius: 10
  },
  circle: {
    borderWidth: 1.5,
    alignItems: "center",
    justifyContent: "center",
    width: 25,
    height: 25,
    borderRadius: 50
  },
  closeIcon: {
    width: 10,
    height: 10,
    resizeMode: "contain"
  },
  flatList: { marginTop: SPACE.xl },
  flatListItem: {
    paddingVertical: 10,
    alignSelf: "center",
    fontSize: FONT_SIZE.lg
  }
});
