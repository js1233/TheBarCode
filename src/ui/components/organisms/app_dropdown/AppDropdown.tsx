import {
  ImageStyle,
  Pressable,
  StyleProp,
  StyleSheet,
  View,
  ViewStyle
} from "react-native";
import React, { useState } from "react";
import { AppLabel } from "ui/components/atoms/app_label/AppLabel";
import { AppLog, SvgProp } from "utils/Util";
import { DropdownModal } from "ui/components/organisms/app_dropdown/DropdownModal";
import { usePreferredTheme } from "hooks";
import { DropDownItem } from "models/DropDownItem";
import ChevronDown from "assets/images/dropdown.svg";
import { TextStyle } from "react-native";
import EIntBoolean from "models/enums/EIntBoolean";
import { useCallback } from "react";
import { useEffect } from "react";
import { SPACE } from "config/Dimens";

export interface AppDropdownProps {
  title?: string;
  items: DropDownItem[] | any;
  preselectedItemString?: string | DropDownItem;
  shouldRunCallbackOnStart?: boolean;
  selectedItemCallback: (item: DropDownItem) => void;
  style?: StyleProp<ViewStyle>;
  textStyle?: StyleProp<TextStyle>;
  dropDownIconStyle?: StyleProp<ImageStyle>;
  dialogBgColor?: string;
  dialogCloseIconStyle?: StyleProp<ImageStyle>;
  dropDownIcon?: SvgProp;
  shouldShowCustomIcon?: boolean;
  isLocked?: EIntBoolean;
  onBlur?: () => void;
}

export const AppDropdown = React.memo<AppDropdownProps>(
  ({
    title,
    preselectedItemString,
    items,
    selectedItemCallback,
    dialogBgColor,
    dropDownIconStyle,
    dialogCloseIconStyle,
    style,
    textStyle,
    dropDownIcon,
    shouldShowCustomIcon = false,
    isLocked = EIntBoolean.FALSE,
    shouldRunCallbackOnStart = true,
    onBlur
  }) => {
    const [modalVisible, setModalVisible] = useState<boolean>(false);
    const [selectedItemText, setSelectedItemText] = useState<
      string | undefined
    >(title);

    useEffect(() => {
      setSelectedItemText(title);
    }, [title]);

    const { themedColors } = usePreferredTheme();

    function openModal() {
      AppLog.log(() => "show modal");
      setModalVisible(true);
    }

    function closeModal() {
      AppLog.log(() => "close modal");
      setModalVisible(false);
    }

    const [selectedItemPosition, setSelectedItemPosition] =
      useState<number>(-1);

    const selectedItem = useCallback(
      (item: DropDownItem | any) => {
        AppLog.log(() => "selectedItem " + item.value);
        setModalVisible(false);
        setSelectedItemText(item.value);
        selectedItemCallback(item);
        setSelectedItemPosition(
          items.findIndex(
            (optionItem: any) => optionItem.value === item.value
          )
        );
      },
      [selectedItemCallback, items]
    );

    // show pre-selected item's text
    useEffect(() => {
      if (selectedItemPosition === -1) {
        let _selectedItemIndex = items.findIndex(
          (item: any) =>
            item.value?.toLowerCase() ===
            preselectedItemString?.toString()?.toLowerCase()
        );

        if (_selectedItemIndex !== -1) {
          selectedItem(items[_selectedItemIndex]);
        }
      }
    }, [
      items,
      selectedItem,
      preselectedItemString,
      shouldRunCallbackOnStart,
      selectedItemPosition
    ]);

    return (
      <View
        style={[
          styles.root,
          {
            backgroundColor: themedColors.secondaryBackground
          },
          style
        ]}>
        <DropdownModal
          isVisible={modalVisible}
          items={items}
          closeModal={() => {
            closeModal();
          }}
          selectedItemCallback={selectedItem}
          dropDownBgColor={dialogBgColor}
          dialogCloseIconStyle={dialogCloseIconStyle}
          selectedItemPosition={selectedItemPosition}
          isLocked={isLocked}
        />

        <Pressable
          testID="dropdown-click"
          onPress={() => {
            onBlur?.();
            !isLocked && openModal();
          }}>
          <View style={[styles.wrapper]}>
            <AppLabel
              text={selectedItemText}
              style={[
                styles.text,
                {
                  color:
                    selectedItemText === title
                      ? themedColors.placeholderColor
                      : themedColors.interface["900"]
                },
                textStyle
              ]}
            />
            {!shouldShowCustomIcon ? (
              <ChevronDown
                width={15}
                height={15}
                fill={themedColors.primaryBackground}
                style={[styles.dropdownIcon, dropDownIconStyle]}
              />
            ) : (
              dropDownIcon?.(themedColors.interface[900])
            )}
          </View>
        </Pressable>
      </View>
    );
  }
);

const styles = StyleSheet.create({
  root: {
    borderRadius: 5,
    justifyContent: "center"
  },
  wrapper: {
    paddingVertical: 10,
    paddingHorizontal: SPACE.md,
    flexDirection: "row",
    justifyContent: "space-between",
    alignItems: "center",
    height: 44
  },
  dropdownIcon: {
    width: 12,
    aspectRatio: 12 / 12,
    resizeMode: "contain"
  },
  text: { includeFontPadding: false }
});
