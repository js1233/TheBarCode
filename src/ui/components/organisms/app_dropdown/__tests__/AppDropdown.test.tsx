import renderer, { ReactTestInstance } from "react-test-renderer";
import React from "react";
import { AppDropdown } from "ui/components/organisms/app_dropdown/AppDropdown";
import {
  fireEvent,
  render,
  RenderAPI,
  waitFor
} from "@testing-library/react-native";
import { DropDownItem } from "models/DropDownItem";

let items: Array<DropDownItem> = [
  {
    value: "Male",
    text: "male"
  },
  {
    value: "Female",
    text: "female"
  },
  {
    value: "Other",
    text: "other"
  }
];

function getDropDownModal(
  getByTestId: (testID: string | RegExp) => ReactTestInstance
) {
  return getByTestId("dropdown-modal");
}

//click on dropdown to open modal
function performDropDownClick(
  getByTestId: (testID: string | RegExp) => ReactTestInstance
) {
  const dropDownClick = getByTestId("dropdown-click");
  fireEvent.press(dropDownClick);
}

function renderItem(onSelectedItem: jest.Mock<any, any>): RenderAPI {
  return render(
    <AppDropdown
      title="Select Gender"
      items={items}
      selectedItemCallback={onSelectedItem}
    />
  );
}

it("Renders snapshot as expected", () => {
  // given
  const onSelectedItem = jest.fn();
  const tree = renderer
    .create(
      <AppDropdown
        title="Select Gender"
        items={items}
        selectedItemCallback={onSelectedItem}
      />
    )
    .toJSON();

  // then
  expect(tree).toMatchSnapshot();
});

describe("check visibility of modal", () => {
  it("check modal visibility when item is not selected", async () => {
    // given
    const onSelectedItem = jest.fn();
    const { getByTestId } = renderItem(onSelectedItem);

    // then
    expect(getDropDownModal(getByTestId).props.visible).toBe(false);

    //when
    performDropDownClick(getByTestId);

    //then
    await waitFor(() => {
      expect(getDropDownModal(getByTestId).props.visible).toBe(true);
    });

    //when
    const closeModal = getByTestId("close-modal");
    fireEvent.press(closeModal);

    //then
    await waitFor(() => {
      expect(getDropDownModal(getByTestId).props.visible).toBe(false);
    });
  });

  it("check modal should invisible when item is selected", async () => {
    // given
    const onSelectedItem = jest.fn();
    const { getByTestId, queryAllByTestId } = renderItem(onSelectedItem);

    // then
    expect(getDropDownModal(getByTestId).props.visible).toBe(false);

    //when
    performDropDownClick(getByTestId);

    //then
    await waitFor(() => {
      expect(getDropDownModal(getByTestId).props.visible).toBe(true);
    });

    //when
    const dropDownItemClick = queryAllByTestId("dropdown-item-click");
    fireEvent.press(dropDownItemClick[0]); //pick first item from dropdown list

    //then
    await waitFor(() => {
      expect(getDropDownModal(getByTestId).props.visible).toBe(false);
    });
  });
});

it("check dropdown shows 3 entries", async () => {
  // given
  const onSelectedItem = jest.fn();
  const { getByTestId, queryAllByTestId } = renderItem(onSelectedItem);

  // then
  expect(getDropDownModal(getByTestId).props.visible).toBe(false);

  //when
  performDropDownClick(getByTestId);

  //then
  await waitFor(() => {
    expect(getDropDownModal(getByTestId).props.visible).toBe(true);
  });

  //then
  const dropDownItems = queryAllByTestId("dropdown-item-click");
  expect(dropDownItems.length).toBe(3);
});

it("check if dropdown shows selected item with right value", async () => {
  // given
  const onSelectedItem = jest.fn();
  const { getByTestId, queryAllByTestId } = renderItem(onSelectedItem);

  //when
  performDropDownClick(getByTestId);
  const dropDownItemClick = queryAllByTestId("dropdown-item-click");
  fireEvent.press(dropDownItemClick[0]); //pick first item from dropdown list

  //then
  await waitFor(() => {
    expect(onSelectedItem).toHaveBeenCalledWith({
      value: "Male",
      text: "male"
    });
  });

  //when
  performDropDownClick(getByTestId);
  fireEvent.press(dropDownItemClick[2]); //pick 3rd item from dropdown list

  //then
  await waitFor(() => {
    expect(onSelectedItem).toHaveBeenLastCalledWith({
      value: "Other",
      text: "other"
    });
  });

  //then
  expect(onSelectedItem).toHaveBeenCalledTimes(2);
});
