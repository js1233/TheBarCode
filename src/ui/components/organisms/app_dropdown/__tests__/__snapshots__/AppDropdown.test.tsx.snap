// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders snapshot as expected 1`] = `
<View
  style={
    Array [
      Object {
        "borderRadius": 5,
        "justifyContent": "center",
      },
      Object {
        "backgroundColor": "#F5F5F5",
      },
      undefined,
    ]
  }
>
  <Modal
    animationType="none"
    hardwareAccelerated={false}
    testID="dropdown-modal"
    transparent={true}
    visible={false}
  >
    <View
      style={
        Object {
          "backgroundColor": "rgba(90,94,94,0.7)",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "flex": 0,
            "height": "auto",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "borderRadius": 10,
                "justifyContent": "center",
                "marginHorizontal": 25,
                "marginVertical": 50,
              },
              Object {
                "backgroundColor": "#F5F5F5",
              },
            ]
          }
        >
          <View
            style={
              Object {
                "marginTop": 18,
              }
            }
          >
            <RCTScrollView
              ListFooterComponent={
                Object {
                  "$$typeof": Symbol(react.memo),
                  "compare": null,
                  "type": [Function],
                }
              }
              data={
                Array [
                  Object {
                    "text": "male",
                    "value": "Male",
                  },
                  Object {
                    "text": "female",
                    "value": "Female",
                  },
                  Object {
                    "text": "other",
                    "value": "Other",
                  },
                ]
              }
              getItem={[Function]}
              getItemCount={[Function]}
              initialNumToRender={3}
              keyExtractor={[Function]}
              onContentSizeChange={[Function]}
              onEndReachedThreshold={1}
              onLayout={[Function]}
              onMomentumScrollBegin={[Function]}
              onMomentumScrollEnd={[Function]}
              onScroll={[Function]}
              onScrollBeginDrag={[Function]}
              onScrollEndDrag={[Function]}
              refreshing={false}
              removeClippedSubviews={true}
              renderItem={[Function]}
              scrollEventThrottle={50}
              showsVerticalScrollIndicator={false}
              stickyHeaderIndices={Array []}
              viewabilityConfigCallbackPairs={Array []}
            >
              <View>
                <View
                  onLayout={[Function]}
                  style={null}
                >
                  <View
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Array [
                        Object {
                          "opacity": 1,
                        },
                      ]
                    }
                    testID="dropdown-item-click"
                  >
                    <Text
                      numberOfLines={1}
                      style={
                        Array [
                          Object {
                            "fontFamily": "ProximaNova-Regular",
                          },
                          Object {
                            "color": "#171717",
                            "fontSize": 27,
                          },
                          Object {
                            "lineHeight": 45,
                          },
                          Array [
                            Object {
                              "alignSelf": "center",
                              "fontSize": 32,
                              "paddingVertical": 10,
                            },
                            Object {
                              "color": "#404040",
                            },
                          ],
                        ]
                      }
                    >
                      Male
                    </Text>
                  </View>
                </View>
                <View
                  onLayout={[Function]}
                  style={null}
                >
                  <View
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Array [
                        Object {
                          "opacity": 1,
                        },
                      ]
                    }
                    testID="dropdown-item-click"
                  >
                    <Text
                      numberOfLines={1}
                      style={
                        Array [
                          Object {
                            "fontFamily": "ProximaNova-Regular",
                          },
                          Object {
                            "color": "#171717",
                            "fontSize": 27,
                          },
                          Object {
                            "lineHeight": 45,
                          },
                          Array [
                            Object {
                              "alignSelf": "center",
                              "fontSize": 32,
                              "paddingVertical": 10,
                            },
                            Object {
                              "color": "#404040",
                            },
                          ],
                        ]
                      }
                    >
                      Female
                    </Text>
                  </View>
                </View>
                <View
                  onLayout={[Function]}
                  style={null}
                >
                  <View
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Array [
                        Object {
                          "opacity": 1,
                        },
                      ]
                    }
                    testID="dropdown-item-click"
                  >
                    <Text
                      numberOfLines={1}
                      style={
                        Array [
                          Object {
                            "fontFamily": "ProximaNova-Regular",
                          },
                          Object {
                            "color": "#171717",
                            "fontSize": 27,
                          },
                          Object {
                            "lineHeight": 45,
                          },
                          Array [
                            Object {
                              "alignSelf": "center",
                              "fontSize": 32,
                              "paddingVertical": 10,
                            },
                            Object {
                              "color": "#404040",
                            },
                          ],
                        ]
                      }
                    >
                      Other
                    </Text>
                  </View>
                </View>
                <View
                  onLayout={[Function]}
                />
              </View>
            </RCTScrollView>
          </View>
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderRadius": 10,
                "justifyContent": "center",
                "left": 25,
                "padding": 10,
                "position": "absolute",
                "top": 50,
              },
              Object {
                "backgroundColor": "#F5F5F5",
              },
            ]
          }
          testID="close-modal"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 50,
                  "borderWidth": 1.5,
                  "height": 25,
                  "justifyContent": "center",
                  "width": 25,
                },
                Object {
                  "backgroundColor": "#E5E5E5",
                  "borderColor": "#D4D4D4",
                },
              ]
            }
          >
            <SvgMock
              fill="#404040"
              height={15}
              width={15}
            />
          </View>
        </View>
      </View>
    </View>
  </Modal>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    testID="dropdown-click"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 44,
            "justifyContent": "space-between",
            "paddingHorizontal": 12,
            "paddingVertical": 10,
          },
        ]
      }
    >
      <Text
        numberOfLines={1}
        style={
          Array [
            Object {
              "fontFamily": "ProximaNova-Regular",
            },
            Object {
              "color": "#171717",
              "fontSize": 27,
            },
            Object {
              "lineHeight": 42,
            },
            Array [
              Object {
                "includeFontPadding": false,
              },
              Object {
                "color": "#737373",
              },
              undefined,
            ],
          ]
        }
      >
        Select Gender
      </Text>
      <SvgMock
        fill="#FFFFFF"
        height={15}
        style={
          Array [
            Object {
              "aspectRatio": 1,
              "resizeMode": "contain",
              "width": 12,
            },
            undefined,
          ]
        }
        width={15}
      />
    </View>
  </View>
</View>
`;
